nvtabular/io.py

```diff
@@ -314,48 +398,47 @@ def _set_dtypes(self, chunk):
                 chunk[col] = chunk[col].astype(dtype)
 
 
-def _shuffle_gdf(gdf, gdf_size=None):
-    """ Shuffles a cudf dataframe, returning a new dataframe with randomly
-    ordered rows """
-    gdf_size = gdf_size or len(gdf)
-    arr = cp.arange(gdf_size)
-    cp.random.shuffle(arr)
-    return gdf.iloc[arr]
-
-
 class Writer:
     def __init__(self):
         pass
 
     def add_data(self, gdf):
         raise NotImplementedError()
 
-    def close(self):
-        pass
-
+    def package_general_metadata(self):
+        raise NotImplementedError()
 
-class Shuffler(Writer):
-    def __init__(
-        self, out_dir, num_out_files=30, num_threads=4, cats=None, conts=None, labels=None
-    ):
-        self.writer = ParquetWriter(out_dir, num_out_files, num_threads, cats, conts, labels)
+    @classmethod
+    def write_general_metadata(cls, data, fs, out_dir):
+        raise NotImplementedError()
 
-    def add_data(self, gdf):
-        self.writer.add_data(_shuffle_gdf(gdf))
+    @classmethod
+    def write_special_metadata(cls, data, fs, out_dir):
+        raise NotImplementedError()
 
     def close(self):
-        self.writer.close()
+        pass
 
 
 class ThreadedWriter(Writer):
     def __init__(
-        self, out_dir, num_out_files=30, num_threads=4, cats=None, conts=None, labels=None
+        self,
+        out_dir,
+        num_out_files=30,
+        num_threads=4,
+        cats=None,
+        conts=None,
+        labels=None,
+        shuffle=None,
+        fs=None,
     ):
         # set variables
+        self.hugectr_bin = False
```

Comment:
Whats this hugectr_bin flag for? Can we remove it?

Reply:
Yeah, sorry - Didn't mean to leave this in!